### ISSUE TITLE
Share the training log 

### ISSUE THREAD

**Issue by hbellafkir** [Audio-WestlakeU/audiossl#11]
Is there any way to share the training log of the trained models? I would like to see how the loss has changed during training and compare it to my training on a custom data set.

**Comment by hbellafkir** [comment]
I should add that I am interested in the training log of the ATST-Frame method

**Comment by lmaxwell** [comment]
Hi @hbellafkir 
Here are the training loss of small model and base model:

- Small
<img width="701" alt="small" src="https://github.com/Audio-WestlakeU/audiossl/assets/1841166/811b4eff-e7e7-4134-9ad0-d2c0bb276e59">

- Base
<img width="698" alt="base" src="https://github.com/Audio-WestlakeU/audiossl/assets/1841166/7de451ba-2369-49eb-ae35-705a3b44e5c8">

The model is prone to model collapse, which is sensitive to learning rate and EMA decay rate.  Besides the loss, you may monitor the std_frm_tea and std_frm_stu.  If both loss and std_frm_* are very small, e.g. loss < 0.1  and std_rm_* << 0.51, the model may collapse and you should use small lerning rate and larger EMA decay rate.

**Comment by hbellafkir** [comment]
Hi @lmaxwell, 

Thank you for your reply. It is an interesting loss curve in the case of the base model. It would be interesting to know if the checkpoint at 100k can achieve similar results as the checkpoint at the 400k iteration. What is actually the criterion to stop the training, since the loss is not yet fully converged in both cases (small and base)? 

I train the small version of ATST and use the pretraining weights for adapting to the birdsong domain with about 600k short soundscapes. Here is the configuration I use:

```
sr = 32000
win_len = 1024      
hop_length = 320
f_max = 15000
warmup_steps= 1950
max_steps = 18000
batch_size = 1024
pos_type = "interpolate"
lr= 2e-5
```
And here is the current state of the training:

![Screenshot from 2024-03-12 10-36-45](https://github.com/Audio-WestlakeU/audiossl/assets/49009668/87fc232b-b943-4caf-a2df-313c258c79c3)

I'm not sure if it's worth continuing to train, is there anything to consider here?

**Comment by lmaxwell** [comment]
Hi, @hbellafkir

> Hi @lmaxwell,
> 
> Thank you for your reply. It is an interesting loss curve in the case of the base model. It would be interesting to know if the checkpoint at 100k can achieve similar results as the checkpoint at the 400k iteration. What is actually the criterion to stop the training, since the loss is not yet fully converged in both cases (small and base)?

Checkpoint at 400K steps performs much better than checkpoint at 100K steps. Unfortunately, there is no criteria to stop the training. Training 400K steps cost 6 days using 7 A100 80G GPU.  Considerting hyper-parmeters tuning, we did not train more steps.  Actually, we have trained small model for 100K steps, obtaining a significant performance boost. 

> I train the small version of ATST and use the pretrained weights for adapting to the birdsong domain with about 600k short soundscapes. Here is the configuration I use:
> 
> ```
> sr = 32000
> win_len = 1024      
> hop_length = 320
> f_max = 15000
> anchor_len = 5 # s
> warmup_steps= 1950
> max_steps = 18000
> batch_size = 1024
> pos_type = "interpolate"
> lr= 2e-5
> ```





> And here is the current state of the training:
> 
> ![Screenshot from 2024-03-12 10-36-45](https://private-user-images.githubusercontent.com/49009668/312027239-87fc232b-b943-4caf-a2df-313c258c79c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMjE0MjMsIm5iZiI6MTcxMDMyMTEyMywicGF0aCI6Ii80OTAwOTY2OC8zMTIwMjcyMzktODdmYzIzMmItYjk0My00Y2FmLWEyZGYtMzEzYzI1OGM3OWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDA5MTIwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4Y2VlNTE4N2U3NTcwMDI5NmEwMjNjODY5NjhmMDJhMjZjMWJlOGVmMjFiYjAwMTE4NDdiMDM3ZjcwMWVkMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vkmhwVa1lDPIvKfrg-dtY_U4W5zAk6OQMZY6wIF1TrQ)
> 
> I'm not sure if it's worth continuing to train, is there anything to consider here?

There seems to be two major  problems:

- The pretrained model was trained using 16K Hz audios.  If you are fintuning using pretrained weights, you should fintune using 16KHz audios too.

- the pos_type should be "cut". In our experiments, "cut" is always better than "interpolate". The advantage of "interpolate" is that the model can process audios audios longer than the pretrained audios. For model trainded using "cut",  audios logner than the pretrained audios can be splitted into chunks, which is exactly the way used by ATST-Frame.

You may check the hyper-parameter setting of the pretrained model. 
```CUDA_VISIBLE_DEVICES=0,1,2,3,4,5 python train.py --arch base \ 
--data_path YOUR_DATA_PATH  \
--save_path YOUR_MODEL_SAVE_PATH \ 
--nproc 6 --batch_size_per_gpu 144 --warmup_steps 19900 \
--max_steps 398000 \
--ema 0.9996 \
--subset 3000000 \
--learning_rate 8e-5 \
--patch_h 64 \
--patch_w 4 \
--mask_type block \
--mask_ratio 0.65 \
--mask_len 5 \
--symmetric True \
--n_mels 64 \
--anchor_len 10 \
--aug_tea False \
--aug_stu True
```

**Comment by hbellafkir** [comment]
Thank you a lot for your feedback @lmaxwell 


> * The pretrained model was trained using 16KHz audios.  If you are fintuning using pretrained weights, you should fintune using 16KHz audios too.


Using 16 kHz for fine tuning is not possible in my case as Bird sounds go up to 15 kHz. I assumed that using the Audioset weights is still better than random initialization. Do you think it is better to train from scratch with 32 KHz?

**Comment by lmaxwell** [comment]
Alright, I get the situation. It is hard to tell which way is better. 

 ATST-small was trained using 100K samples, and ATST-Frame used 2 million samples. Since you have 600K training samples, I think it is sufficent to train a small model and may be sufficent for a base model too.

 For finetuing using 32K Hz,  it is worth mentioning that some parts of the code in transform.py are hard-coded to work with 16K Hz audios, which you may adapt to work with 32K Hz audios.  

> Thank you a lot for your feedback @lmaxwell
> 
> > * The pretrained model was trained using 16KHz audios.  If you are fintuning using pretrained weights, you should fintune using 16KHz audios too.
> 
> Using 16 kHz for finetuning is not possible in my case as Bird sounds go up to 15 kHz. I assumed that using the Audioset weights is still better than random initialization. Do you think it is better to train from scratch with 32 KHz?
